homeAndDetailsView

gonzalorome7 opened the pull request:
both ready views, the view model itself is missing for the detail view and the previous and next buttons.

gonzalorome7 committed:
homeAndDetailsView

nicolasdemaioglobant commented on TedTalkMVVM/TedTalkMVVM/DisplayManagerTedTalk.swift:
```diff
@@ -0,0 +1,44 @@
+}
+class DisplayManagerTedTalk {
+    
+    static let shareTedTalkManager: DisplayManagerTedTalk = DisplayManagerTedTalk()
```
typo here, it should be shared instead of share

nicolasdemaioglobant replied:
Also to make this class a singleton you should make init method as private, otherwise from outside could create several instances of DisplayManagerTedTalk

nicolasdemaioglobant commented on TedTalkMVVM/TedTalkMVVM/view/DetailTedTalkViewController.swift:
```diff
@@ -0,0 +1,41 @@
+            nameDetailLabel.text = displayTedTalk.name
+            descriptionDetailLabel.text = displayTedTalk.description
+        }
+    }
```
remember this detail view controller should have 2 new buttons to get next and previous ted talk

nicolasdemaioglobant commented on TedTalkMVVM/TedTalkMVVM/view/HomeViewController.swift:
```diff
@@ -0,0 +1,84 @@
+    @IBOutlet weak var searchTedTalk: UISearchBar!
+    
+    var viewModel: ViewModelTedTalkProtocol!
+    var tedTalksDisplay: [TedTalkDisplay] = []
```
these properties could be private

nicolasdemaioglobant commented on TedTalkMVVM/TedTalkMVVM/view/HomeViewController.swift:
```diff
@@ -0,0 +1,84 @@
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        viewModel.getPickerRowsCount()
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        viewModel.getPickerRow(index: row)
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        pickerSelectedRow = viewModel.getPickerRow(index: row)
+    }
```
instead of asking to the viewModel to get picker data you should implement an approach similary as you did for the table view

gonzalorome7 committed:
ViewModelDetailsAndTwoButtons

make corrections, create the new view model for the detail view and the two missing buttons

gonzalorome7 committed:
ButtonsEnablesAndDisables

nicolasdemaioglobant commented on TedTalkMVVM/TedTalkMVVM/view/DetailTedTalkViewController.swift:
```diff
@@ -16,26 +16,56 @@ class DetailTedTalkViewController: UIViewController{
+        viewModel = viewModelDetail(tedTalkManager: DisplayManagerTedTalk.sharedTedTalkManager,
+                                    observerTedTalk: { [unowned self] viewModel in
+                                        self.showTedTalkDetail(talk: viewModel.getTedTalkDetailsDisplay())
+                                    },observerNextTedTalkButton: { [unowned self] viewModel in
+                                        self.buttonNextDisable()}, observerPreviousTedTalkButton: { [unowned self] viewModel in
+                                            self.buttonPreviousDisable()}, observerTedTalkButtons: { [unowned self] viewModel in
+                                                self.buttonsEnables()})
```
The closure for each button (next & previous) should return a boolean as a parameter, to indicate to the view if that button has to be enabled or not.
Also I do not think you need the third closure (observerTedTalkButtons), because take into consideration some border cases like index 0, which in that case only next button should be enabled, not previous and in the last ted talk of the array only previous button should be enabled not next one.
As it is now, you are enabling both buttons at the same time

gonzalorome7 committed:
changueEnableButtons

nicolasdemaioglobant commented on TedTalkMVVM/TedTalkMVVM/view/DetailTedTalkViewController.swift:
```diff
@@ -37,10 +37,8 @@ class DetailTedTalkViewController: UIViewController{
-                                        self.buttonNextDisable()}, observerPreviousTedTalkButton: { [unowned self] viewModel in
-                                            self.buttonPreviousDisable()}, observerTedTalkButtons: { [unowned self] viewModel in
-                                                self.buttonsEnables()})
+                                    }, observerTedTalkButtons: { [unowned self] viewModelPrevious, viewModelNext  in
```
just a minor comment, instead of naming those variables like viewModelPrevious it could be better to name them something like: previousButtonIsActive, it will be much more clear

gonzalorome7 committed:
ChangueLittleDetailesNameVariable
